### ISSUE TITLE
bug w.r.t. to multicolumn masking

### ISSUE THREAD

**Issue by jprobichaud** [RWKV/RWKV-infctx-trainer#35]
https://github.com/RWKV/RWKV-infctx-trainer/blob/d97c2b94eadb34cb73b2e8b1fcfcf0e3e9c7b829/RWKV-v5/src/data.py#L296

The statement here is wrong, it should be 
```
    if len(multi_column_train_mask) > i and multi_column_train_mask[i] is False:
````

**Comment by m8than** [comment]
I believe the original is correct, it just means if the mask variable for the column is set and is false then mask it from the loss value.

**Comment by jprobichaud** [comment]
no, the left part of this condition is mostly always false so the condition is skipped

**Comment by PicoCreator** [comment]
If the left part is always false, it means the `multi_column_train_mask` was not configured.
Which should be the same length as `multi_column_keys` (which is being iterated for the i loop)

I think the confusion here is regarding "What should be the default behaviour" if the `multi_column_train_mask` is not configured, which was not defined

**Comment by PicoCreator** [comment]
Im changing to the following for v5
```
# Configure the attention masks accordingly
if i > len(multi_column_train_mask):
    # If the `multi_column_train_mask` is not set, we will assume all columns are training data
    attention_mask += ([1] * len(column_encodings['input_ids']))
elif multi_column_train_mask[i] is False:
    # If the `multi_column_train_mask` is set, but configured as false, we should not pay attention to it
    attention_mask += ([0] * len(column_encodings['input_ids']))
else: # multi_column_train_mask[i] is True
    # This means it is true, lets pay attention once again
    attention_mask += ([1] * len(column_encodings['input_ids']))
```
